src/materials/pathtracing/glsl/directLightContribution.glsl.js

```diff
@@ -78,15 +80,15 @@ export const directLightContributionGLSL = /*glsl*/`
 					// weight the direct light contribution
 					envPdf /= lightsDenom;
 					float misWeight = misHeuristic( envPdf, envMaterialPdf );
-					return attenuatedColor * environmentIntensity * envColor * state.throughputColor * sampleColor * misWeight / envPdf;
+					result = attenuatedColor * environmentIntensity * envColor * state.throughputColor * sampleColor * misWeight / envPdf;
 
 				}
 
 			}
 
 		}
 
-		return vec3( 0.0 );
+		return result;
```

Comment:
For the sake of documenting some of these decisions can we add a comment saying that the Mac regressions seem to be related to early return statements? Maybe with a reference to this PR id, as well.